app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -387,6 +396,35 @@ open class HomeActivity : LocaleAwareAppCompatActivity() {
         return DefaultThemeManager(browsingModeManager.mode, this)
     }
 
+    @ExperimentalCoroutinesApi
+    private fun performanceTestingIntentProcessing(intent: Intent) {
+
+        var performanceTestingOn = false
+
+        if (intent.getBooleanExtra(EXTRA_PERFORMANCE_TEST, false)) {
+            val batteryStatus = baseContext.registerReceiver(null, IntentFilter(Intent.ACTION_BATTERY_CHANGED))
+            val isPhonePlugged = batteryStatus!!.getIntExtra(BatteryManager.EXTRA_PLUGGED, -1)
+            val isAdbenabled = Settings.Global.getInt(baseContext.getContentResolver(), Settings.Global.ADB_ENABLED, 0)
+
+            performanceTestingOn = (isPhonePlugged == BatteryManager.BATTERY_PLUGGED_USB) && (isAdbenabled == 1)
+        }
+        if (intent.getBooleanExtra(EXTRA_FINISH_ONBOARDING, false) && performanceTestingOn) {
+            FenixOnboarding(this).finish()
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+            intent.getBooleanExtra(EXTRA_VISUAL_METRICS, false) && performanceTestingOn) {
+            window.decorView.foreground = ColorDrawable(Color.BLUE)
+            val url = intent.getStringExtra(URL_TO_NAVIGATE)
+            supportFragmentManager.registerFragmentLifecycleCallbacks(VisualMetricsInstrumentation(url), true)
+        }
+        if (intent.getBooleanExtra(EXTRA_NO_TP, false) && performanceTestingOn) {
+            MainScope().launch(Dispatchers.IO) {
+                settings().incrementTrackingProtectionOnboardingCount()
```

Comment:
nit: why twice? add a comment.

This seems fragile: is there a better way to do this? Maybe it's better to address this when it actually becomes a problem and breaks page load testing in the future because we shouldn't spend too much more time on this. :)